keyboards/h87a/keymaps/gam3cat/keymap.c

```diff
@@ -0,0 +1,288 @@
+    *    |      |   |   |_DL|   |   |   |   |   |   |   |   |        |           |
+    *    |-----------------------------------------------------------|-----------|
+    *    |        |   |   |_CL|   |_BL|_NL|   |   |   |   |      |Fn |   |   |   |
+    *    |-----------------------------------------------------------|-----------|
+    *    |     |   |     |                           |     |Fn |     |   |   |   |
+    *    *-----------------------------------------------------------------------*
+    */
+    [_AL] = LAYOUT_all( \
+        RESET,            XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,          XXXXXXX, XXXXXXX, XXXXXXX, \
+        QMK_REV, RGB_TOG, RGB_MOD, RGB_HUD, RGB_HUI, RGB_SAD, RGB_SAI, RGB_VAD, RGB_VAI, XXXXXXX, BL_TOGG, BL_DEC,  BL_INC,  XXXXXXX, XXXXXXX, XXXXXXX, KC_DMR1, KC_DMP1, \
+        _______, DF(_BL), DF(_WL), XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,          KC_DMRS, KC_DMR2, KC_DMP2, \
+        XXXXXXX, XXXXXXX, XXXXXXX, DF(_DL), XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                                     \
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, DF(_CL), XXXXXXX, DF(_BL), DF(_NL), XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, _______,                   XXXXXXX,          \
+        XXXXXXX, XXXXXXX, XXXXXXX,                            XXXXXXX,                            XXXXXXX, _______, XXXXXXX, XXXXXXX,          XXXXXXX, XXXXXXX, XXXXXXX \
+    ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case QMK_REV:
+            if (record->event.pressed) {
+                SEND_STRING(QMK_KEYBOARD "/" QMK_KEYMAP "@" QMK_VERSION ":" QMK_BUILDDATE);
+            }
+            return false;
+            break;
+        case KC_WEB:
+            if (record->event.pressed) {
+                SEND_STRING(SS_LGUI("r"));
+                _delay_ms(100);
+                SEND_STRING("chrome.exe\n");
+            }
+            return false;
+            break;
+        case KC_SP4:
+            if (record->event.pressed) {
+                SEND_STRING ("    ");
+            }
+            return false;
+            break;
+    }
+    // Dynamic Macros.
+    if (!process_record_dynamic_macro(keycode, record)) {
+        return false;
+    }
+    return true;
+}
+
+void matrix_init_user(void) {
+    #ifdef BACKLIGHT_ENABLE
+        backlight_level(0);
+    #endif
+    #ifdef RGBLIGHT_ENABLE
+        rgblight_mode(1);
+        rgblight_sethsv(180,100,100);
+    #endif
+}
+
+void matrix_scan_user(void) {
+    uint8_t layer = biton32(layer_state);
+
+    if (current_layer == layer) {
+    }
+    else {
+        current_layer = layer;
+        switch (current_layer) {
+            case _BL:
+                backlight_level(0);
+                rgblight_sethsv(180,100,255);
```

Comment:
Same as above, but you should DEFINITELY not call these in the matrix scan.  You may have noticed slowdown in your keymaps when doing this.

All of this code should be called in `layer_state_set_user`, and you should use `rgblight_sethsv_noeeprom`